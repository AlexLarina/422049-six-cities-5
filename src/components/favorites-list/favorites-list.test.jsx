import React from "react";
import renderer from "react-test-renderer";
import FavoritesList from "./favorites-list.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";

const offer = {
  id: 0,
  premium: true,
  photo: ``,
  cost: 0,
  title: ``,
  type: ``,
  rating: 0,
  bedrooms: 0,
  maxGuests: 0,
  owner: {
    name: ``
  },
  household: [],
  images: [],
  neighbors: [],
  coordinates: [],
  description: ``
};

const mockStore = configureStore()({
  DATA: {
    favoriteOfferList: []
  }
});

it(`Render FavoritesList`, () => {
  const tree = renderer
    .create(
        <Provider store={mockStore}>
          <MemoryRouter>
            <FavoritesList
              offer={offer}
              favoriteOfferList={[]}
            />
          </MemoryRouter>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
