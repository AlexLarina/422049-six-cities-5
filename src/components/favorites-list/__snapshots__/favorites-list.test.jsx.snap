// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FavoritesList 1`] = `
<section
  className="favorites"
>
  <h1
    className="favorites__title"
  >
    Saved listing
  </h1>
  <ul
    className="favorites__list"
  >
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      />
    </li>
  </ul>
</section>
`;
